src/pydvl/utils/parallel/map_reduce.py

```diff
@@ -226,9 +226,7 @@ def _backpressure(
         self, jobs: List[ObjectRef], n_dispatched: int, n_finished: int
     ) -> int:
         if (n_in_flight := n_dispatched - n_finished) > self.max_parallel_tasks:
-            wait_for_num_jobs = max(1, self.max_parallel_tasks // 2)
-            # num_returns cannot be bigger than length of provided list
-            wait_for_num_jobs = min(wait_for_num_jobs, n_in_flight)
+            wait_for_num_jobs = max(1, n_in_flight - self.max_parallel_tasks)
```

Comment:
If `n_in_flight > max parallel tasks`, then the difference will always be >=1 and this `min()` is unnecessary.

But more importantly, if below the `timeout` is set, then `ray.wait()` might return before `wait_for_num_jobs` jobs are done. Shouldn't we check instead the returned lists to see the actual number of finished jobs?

Reply:
Nice catch! I will fix it right away